src/main/java/uk/ac/ebi/eva/evaseqcol/dus/AssemblySequencesReader.java

```diff
@@ -0,0 +1,38 @@
+package uk.ac.ebi.eva.evaseqcol.dus;
+
+import uk.ac.ebi.eva.evaseqcol.entities.AssemblySequencesEntity;
+
+import java.io.BufferedReader;
+import java.io.IOException;
+import java.io.InputStreamReader;
+import java.security.NoSuchAlgorithmException;
+
+public abstract class AssemblySequencesReader {
+    protected final BufferedReader reader;
+
+    protected final String accession;
+
+    protected AssemblySequencesEntity assemblySequencesEntity;
+
+
+    protected boolean fileParsed = false;
+
+
+    public AssemblySequencesReader(InputStreamReader inputStreamReader, String accession){
+        this.reader = new BufferedReader(inputStreamReader);
```

Comment:
**Optional:** This reader object seems to be closed manually in the inheritees.Check if it is possible to provide a top-level method here that will call parseFile in the inheritees (a method prone to exceptions) and close the reader after parseFile exits.